analyzer/pom-analyzer/src/main/java/eu/fasten/analyzer/pomanalyzer/pom/DataExtractor.java

```diff
@@ -511,11 +543,17 @@ public DependencyData extractDependencyData(String groupId, String artifactId, S
         return dependencies;
     }
 
-    private Optional<String> downloadPom(String artifactId, String groupId, String version)
-            throws FileNotFoundException {
-        for (var repo : this.getMavenRepos()) {
+    private Optional<String> downloadPom(String artifactId, String groupId, String version) {
+        for (var repo : this.mavenRepos) {
             var pomUrl = this.getPomUrl(artifactId, groupId, version, repo);
-            var pom = httpGetToFile(pomUrl).flatMap(DataExtractor::fileToString);
+            Optional<String> pom;
+            try {
+                pom = httpGetToFile(pomUrl).flatMap(DataExtractor::fileToString);
+            } catch (FileNotFoundException | UnknownHostException | MalformedURLException e) {
+                continue;
+            } catch (IOException e) {
+                throw new RuntimeException("Error downloading file: " + e.getMessage());
+            }
```

Comment:
Now when I think about it there might be an IO problem downloading from one repo, but we shouldn't immediately throw a RuntimeException, as there is a chance that it will be ok with other repos. I remember I suggested doing it this way during the conversation, but now when I see the code I think that this might be a problem.

Reply:
How about now? In case of `IOException` the `httpGetToFile()` will return `Optional.empty()` and in the loop in `downloadPom()` it will just proceed to the next repo.